judgments/views/advanced_search.py

```diff
@@ -1,179 +1,188 @@
+                query_params = query_params | {
+                    "from": from_date,
+                    "from_date_0": from_date.day,
+                    "from_date_1": from_date.month,
+                    "from_date_2": from_date.year,
+                }
+            # If a to_date is not specified, set it to the current min year
+            if not to_date:
+                to_date = date(CourtDates.max_year(), 12, 31)
+            else:
+                # Only provide the param back to the user if they set it
+                query_params = query_params | {
+                    "to": to_date,
+                    "to_date_0": to_date.day,
+                    "to_date_1": to_date.month,
+                    "to_date_2": to_date.year,
+                }
+
+            query_params = query_params | {
+                "query": query_text,
+                "courts": form.cleaned_data.get("courts", []),
+                "judge": form.cleaned_data.get("judge", ""),
+                "party": form.cleaned_data.get("party", ""),
+                "order": order,
+            }
+            page: str = str(as_integer("1", minimum=1))
+            # Merge the courts and tribunals as they are treated as the same in MarkLogic.
+            courts_and_tribunals = form.cleaned_data.get(
+                "courts"
+            ) + form.cleaned_data.get("tribunals")
+            # Construct the search parameter object required for Marklogic query
+            search_parameters: SearchParameters = SearchParameters(
+                # Should process query be moved to a clean method?
+                query=query_text,
+                court=",".join(courts_and_tribunals),
+                judge=form.cleaned_data.get("judge_name", ""),
+                party=form.cleaned_data.get("party_name", ""),
+                page=page,
+                order=params.get("order", "-date"),
+                date_from=from_date,
+                date_to=to_date,
+                page_size=int(params.get("per_page", "10")),
             )
 
-            court_facets = {}
-            year_facets = {}
+            # Get the response from Marklogic
+            try:
+                search_response: SearchResponse = search_judgments_and_parse_response(
+                    api_client, search_parameters
+                )
+            except MarklogicResourceNotFoundError:
+                raise Http404("Search failed")
 
+            # If a query was provided, get relevant search facets to display to the user
             if search_parameters.query:
                 unprocessed_facets, court_facets = process_court_facets(
-                    search_response.facets, context["query_params"].get("court")
+                    search_response.facets, form.cleaned_data.get("courts", [])
                 )
                 unprocessed_facets, year_facets = process_year_facets(
                     unprocessed_facets
                 )
+            if from_date:
```

Comment:
Tempted to suggest removing this if -- _do_dates checks for presence, I believe.